### ISSUE TITLE
Can't install  PPA package on Ubuntu 19.10 

### ISSUE THREAD

**Issue by AdnanHodzic** [LordAmit/Brightness#143]
Hi,

`brightness-controller` package from `ppa:apandada1/brightness-controller` PPA can't be installed due to a missing dependency `sni-qt` which is not available for Ubuntu 19.10 (Eoan Ermine).

Exact error I get is:

```
ahodzic@carbon7 ~$ sudo apt-get install brightness-controller
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 brightness-controller : Depends: sni-qt but it is not installable
E: Unable to correct problems, you have held broken packages.
```

Last release of this package can be found in is Ubuntu 19.04 (Disco Dingo): https://packages.ubuntu.com/search?keywords=sni-qt&searchon=names&suite=disco&section=all

I'm not sure what the `sni-qt` replacement package is in 19.10, but list of dependencies for this release should be updated.

**Comment by LordAmit** [comment]
Wow,

Glad to know that users want to continue use Brightness Controller even in 19.10 👍 

@apandada1 - Can you please look into this? 🙏

**Comment by apandada1** [comment]
What does sni-qt do? Does it manage the system tray?

On Sun, Dec 1, 2019, 5:04 AM Amit Seal Ami <notifications@github.com<mailto:notifications@github.com>> wrote:

Wow,

Glad to know that users want to continue use Brightness Controller even in 19.10 👍

@apandada1<https://github.com/apandada1> - Can you please look into this? 🙏

**Comment by LordAmit** [comment]
Uh, you might want to check the email body before sending @apandada1  - Github messed things up and included your unsubscribe link.

About sni-qt, yes. I believe it is being used for the minimize to systray part.

**Comment by apandada1** [comment]
@LordAmit Then will Brightness Controller even work if we remove the dependency? 
@AdnanHodzic Can you download the source (https://github.com/LordAmit/Brightness/archive/master.zip) and report whether it works in Ubuntu 19.10 without sni-qt?

**Comment by apandada1** [comment]
`sni-qt` [got removed](https://askubuntu.com/questions/1193152/how-to-install-sni-qt-in-ubuntu-19-10) from Ubuntu because it is based on QT4, which is no longer supported.
I tried to build it in Launchpad in a ppa, but the server cannot build it due to missing qt4 dependencies. I will try to build it in 19.04 and copy the package in 19.10. I am not sure whether that will work either.

**Comment by apandada1** [comment]
@AdnanHodzic I have made this ppa for sni-qt. If you get dependency issues while installing sni-qt, let us know, I will try to include those dependencies in the ppa as well. (**Warning**: this might break your system, and so don't do this unless you know how to remove a broken package with missing dependencies). If everything works, we can add these packages in the Brightness Controller PPA.

**Comment by AdnanHodzic** [comment]
@apandada1 no worries about breaking my system, I used to be a Debian package maintainer :)

Unfortunately, when trying to install `sni-qt` (sni-qt_0.2.7+17.04.20170112-0_106_ubuntu19.04.1_amd64.deb) from your new PPA, I get an error when it tries to install other dependencies, i.e:

```
Unpacking sni-qt:amd64 (0.2.7+17.04.20170112-0~106~ubuntu19.04.1) ...
dpkg: dependency problems prevent configuration of sni-qt:amd64:
 sni-qt:amd64 depends on libdbusmenu-qt2 (>= 0.3.2); however:
  Package libdbusmenu-qt2 is not installed.

dpkg: error processing package sni-qt:amd64 (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 sni-qt:amd64
```

**Comment by apandada1** [comment]
Sorry for the delay. I have uploaded `libdbusmenu-qt2` to the sni-qt ppa (https://launchpad.net/~apandada1/+archive/ubuntu/sni-qt/). Run `apt update` and let us know whether you can now install `sni-qt` in ubuntu 19.10.

**Comment by apandada1** [comment]
I installed 19.10 in a VM. `sni-qt` is now installable from my PPA after adding `libdbusmenu-qt`. I am copying the package to the main PPA so that `sni-qt` is automatically installed in the future.
![Screenshot 2020-01-12 01:19:25](https://user-images.githubusercontent.com/3063132/72210000-3468ff80-34db-11ea-8956-1ee8f916401b.png)

@LordAmit This issue is now **fixed**. _However_, I saw in Ask Ubuntu that support for qt4 will be completely removed in 20.04. Then this method may not work (unless all relevant qt4 libraries and dependencies are made available in the PPA). Any news for v3?

**Comment by apandada1** [comment]
`python-pyside` is not available in 20.04 anymore (installed in VM). I will try to add it to the ppa in the same way, but I am not sure whether it will work, as it has a huge number of dependencies.

However, `python3-pyside` is available in 20.04. So, it may be easier to port the code to python 3, instead of rewriting.

**Comment by LordAmit** [comment]
@apandada1  - please post a reply here once we are done with porting Brightness Controller to PySide2 and is available in PPA. 😄

**Comment by apandada1** [comment]
I have manually fixed this for 19.10 by uploading sni-qt to the ppa. Pyside2 version is made available for 20.04 only.
